packages/terser/src/index.js

```diff
@@ -1,8 +1,63 @@
 #!/usr/bin/env node
 
-// Pass-through require, ensures that the nodejs_binary will load the version of terser
-// from @bazel/terser package.json, not some other version the user depends on.
-require('terser/bin/uglifyjs');
+const fs = require('fs');
+const path = require('path');
+const child_process = require('child_process');
 
-// TODO: add support for minifying multiple files (eg. a TreeArtifact) in a single execution
-// Under Node 12 it should use the worker threads API to saturate all local cores
\ No newline at end of file
+// Run Bazel with --define=VERBOSE_LOGS=1 to enable this logging
+const VERBOSE_LOGS = !!process.env['VERBOSE_LOGS'];
+
+function log_verbose(...m) {
+  if (VERBOSE_LOGS) console.error('[terser/index.js]', ...m);
+}
+
+// Peek at the arguments to find any directories declared as inputs
+let argv = process.argv.slice(2);
+// terser_minified.bzl always passes the inputs first,
+// then --output [out], then remaining args
+// We want to keep those remaining ones to pass to terser
+// Avoid a dependency on a library like minimist; keep it simple.
+const outputArgIndex = argv.findIndex((arg) => arg.startsWith('--'));
+
+const inputs = argv.slice(0, outputArgIndex);
+const output = argv[outputArgIndex + 1];
+const residual = argv.slice(outputArgIndex + 2);
+
+log_verbose(`Running terser/index.js
+  inputs: ${inputs}
+  output: ${output}
+  residual: ${residual}`);
+
+function isDirectory(input) {
+  return fs.lstatSync(path.join(process.cwd(), input)).isDirectory();
+}
+
+function terserDirectory(input) {
+  if (!fs.existsSync(output)) {
+    fs.mkdirSync(output);
+  }
+
+  fs.readdirSync(input).forEach(f => {
+    if (f.endsWith('.js')) {
+      const inputFile = path.join(input, path.basename(f));
+      const outputFile = path.join(output, path.basename(f));
+      // We don't want to implement a command-line parser for terser
+      // so we invoke its CLI as child processes, just altering the input/output
+      // arguments. See discussion: https://github.com/bazelbuild/rules_nodejs/issues/822
+      // TODO: under Node 12 it should use the worker threads API to saturate all local cores
+      child_process.fork(
```

Comment:
We do this the same as for a single file when under threads, just need to modify process.argv inside the thread

Reply:
okay, good to know, I think I'll try to land this super-simple and let you do the concurrency limiting and hook it up with worker_threads API?

Reply:
Yeah sounds like a plan